src/stm32f1/pwm.cpp

```diff
@@ -0,0 +1,230 @@
+#include "pin.hpp"
+#include "power.hpp"
+#include "pwm_reg.hpp"
+#include <cmath>
+#include <libhal-arm-mcu/stm32f1/clock.hpp>
+#include <libhal-arm-mcu/stm32f1/constants.hpp>
+#include <libhal-arm-mcu/stm32f1/pin.hpp>
+#include <libhal-arm-mcu/stm32f1/pwm.hpp>
+#include <libhal-util/bit.hpp>
+
+namespace hal::stm32f1 {
+namespace {
+[[nodiscard]] pwm_reg_t* get_pwm_reg(peripheral p_id)
+{
+  if (p_id == peripheral::timer1) {
+    return pwm_timer1;
+  }
+  return pwm_timer8;
+}
```

Comment:
Are the only timers that support PWM timer 1 or 8? Do the pins for timer 1 work with timer 8?